crates/core/tedge/src/cli/certificate/renew.rs

```diff
@@ -28,28 +32,30 @@ impl Command for RenewCertCmd {
 
 impl RenewCertCmd {
     fn renew_test_certificate(&self, config: &NewCertificateConfig) -> Result<(), CertError> {
-        let id = cn_of_self_signed_certificate(&self.cert_path)?;
+        let cert_path = &self.cert_path;
+        let key_path = &self.key_path;
+        let id = cn_of_self_signed_certificate(cert_path)?;
 
         // Remove only certificate
         std::fs::remove_file(&self.cert_path)
             .map_err(|e| CertError::IoError(e).cert_context(self.cert_path.clone()))?;
 
         // Re-create the certificate from the key, with new validity
-        let create_cmd = CreateCertCmd {
-            id,
-            cert_path: self.cert_path.clone(),
-            key_path: self.key_path.clone(),
-            csr_path: None,
-            bridge_location: self.bridge_location,
-        };
+        let previous_key = reuse_private_key(key_path)
+            .map_err(|e| CertError::IoError(e).key_context(key_path.clone()))?;
+        let cert = KeyCertPair::new_selfsigned_certificate(config, &id, &previous_key)?;
 
-        create_cmd.renew_test_certificate(config)
+        override_public_key(cert_path, cert.certificate_pem_string()?)
```

Comment:
The explicit removal of the certificate in line. 40 seems redundant because of the "override" mechanism. But not harm either.

Reply:
I thought the same but it is not. Indeed, a certificate is in practice write protected (`0o444`).